### PR TITLE
Match some more frame functions

### PR DESCRIPTION
**Pull request by cadmic**
Unfortunately tons of `#ifndef NON_MATCHING` because of floats

### CODE REVIEW

File: src/emulator/frame.c [hunk 1]
```diff
@@ -2701,14 +2888,94 @@ static inline bool frameCopyMatrix(Mtx44 matrixTarget, Mtx44 matrixSource) {
     return true;
 }
 
-bool frameScaleMatrix(Mtx44 matrixResult, Mtx44 matrix, f32 rScale);
-#pragma GLOBAL_ASM("asm/non_matchings/frame/frameScaleMatrix.s")
+bool frameScaleMatrix(Mtx44 matrixResult, Mtx44 matrix, f32 rScale) {
```
Comment (Yanis42) [inline src/emulator/frame.c:2891]: any reason why they wouldn't use a loop here? (just wondering)

File: src/emulator/frame.c [hunk 2]
```diff
@@ -2783,9 +3095,156 @@ static inline bool frameResetCache(Frame* pFrame) {
     return true;
 }
 
-#pragma GLOBAL_ASM("asm/non_matchings/frame/frameUpdateCache.s")
+static bool frameUpdateCache(Frame* pFrame) {
+    // s32 nCount;
+    // s32 nCountFree;
```
Comment (Yanis42) [inline src/emulator/frame.c:3100]: is this intended?

**Review by Yanis42 — COMMENTED** [review]
nice!

File: src/emulator/frame.c (hunk 2 above)
Comment (cadmic) [reply src/emulator/frame.c:3100]: yeah, it's more variables in the debug info that I'm not sure how to use

File: src/emulator/frame.c (hunk 1 above)
Comment (cadmic) [reply src/emulator/frame.c:2891]: if I had to guess, maybe not trusting the compiler to unroll the loops?

**Review by Yanis42 — APPROVED** [review]

**Pull request merged**